File: frontend/angular-secretroom-website/src/app/model/purchase.ts
import {Customer} from "./customer";
import {Address} from "./address";
import {Order, OrderItem} from "./order";

export class Purchase {

  customer: Customer;
  shippingAddress: Address;
  order: Order;
  orderItems: OrderItem[];
  comment: string;
  language: string;

  constructor(customer: Customer, shippingAddress: Address, order: Order, orderItems: OrderItem[], comment: string, language: string) {
    this.customer = customer;
    this.shippingAddress = shippingAddress;
    this.order = order;
    this.orderItems = orderItems;
    this.comment = comment;
    this.language = language;
  }

}

File: backend/src/main/java/com/secretroomwebsite/purchase/Purchase.java
package com.secretroomwebsite.purchase;
import com.secretroomwebsite.adress.Address;
import com.secretroomwebsite.customer.Customer;
import com.secretroomwebsite.order.Order;
import com.secretroomwebsite.order.OrderItem;
import lombok.Data;
import java.util.List;


@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Order order;
    private List<OrderItem> orderItems;
    private String comment;
    private String language;

}
